bindings/rascal/models/krr.py

```diff
@@ -433,6 +593,19 @@ def train_gap_model(
     jitter : double, optional
         small jitter for the numerical stability of solving the linear system,
         by default 1e-8
+    solver: string, optional
+        method used to solve the sparse KRR equations.
+        "Normal" uses scipy.linalg.solve for the normal equations:
+           (K_NM.T@K_NM + K_MM)^(-1) K_NM.T@Y
+        "lstsq" uses numpy.linalg.solve( rcond=-1) for the normal equations:
+           (K_NM.T@K_NM + K_MM)^(-1) K_NM.T@Y
+        "RKHS" computes first the reproducing kernel features by diagonalizing K_MM
+        and computing P_NM = K_NM @ U_MM @ Lam_MM^(-1.2) and then solves the linear
+        problem for those (which is usually better conditioned)
+           (P_NM.T@P_NM + 1)^(-1) P_NM.T@Y
+        "RKHS-QR"
+        "QR"
+        by default, "RKHS"
```

Comment:
I think we should leave the default at lstsq for now, since that's what we used most of the time. Alternatively, I could see an argument for QR, which is what QUIP is using, and should be more stable. RKHS solvers seems more experimental

Reply:
Just for information: according to michele the RKHS-QR solver is the most stable, then the RKHS, then the QR.

I would add it in the documentation that the QR solver is the same as in QUIP, but I am fine with any solver as the default one. I can see an argument in each case.

Reply:
My point is we want this to be a small & quick PR adding the option to use other solvers so that people using librascal can have model that learn, and for that, I think we need to have the smallest set of changes possible. 

Changing the default can be done later, or in equisolve/equilearn once we understand what's going on.